api/src/users/services/users.service.ts

```diff
@@ -1,85 +1,140 @@
 import {
     Injectable,
     ConflictException,
-    NotFoundException
+    NotFoundException,
+    ForbiddenException
 } from "@nestjs/common";
 import { InjectRepository } from "@nestjs/typeorm";
-import { Repository } from "typeorm";
+import { EntityNotFoundError, QueryFailedError, Repository } from "typeorm";
 import { User } from "users/model/users.entity";
 import { UpdateUserDto } from "users/model/dto/update-user.dto";
 import { UserCredentialsDto } from "auth/model/dto/user-credentials.dto";
+import { ConfigService } from "@nestjs/config";
+import { UsersAvatarService } from "usersAvatar/services/usersAvatar.service";
+import IUploadedFile from "users/types/uploaded-file.interface";
 
 @Injectable()
 export class UsersService {
+    private readonly isDevelopment: boolean;
+
     constructor(
         @InjectRepository(User)
-        private readonly userRepository: Repository<User>
-    ) {}
+        private readonly userRepository: Repository<User>,
+        private readonly configService: ConfigService,
+        private readonly usersAvatarService: UsersAvatarService
+    ) {
+        this.isDevelopment =
+            configService.get("DEVELOPMENT") === "true" || false;
+    }
 
     async create(userCredentialsDto: UserCredentialsDto) {
-        const user = await this.userRepository.findOneBy({
-            username: userCredentialsDto.username
-        });
+        const user = await this.userRepository
+            .findOneBy({
+                username: userCredentialsDto.username
+            })
+            .catch(error => {
+                throw error;
+            });
 
         if (user) throw new ConflictException("Duplicated username");
 
         return this.userRepository.save(userCredentialsDto);
     }
 
     async findAll() {
-        return this.userRepository.find({
-            relations: {
-                userAvatar: true
-            }
-        });
+        return this.userRepository.find();
     }
 
     async findOne(id: number) {
-        return this.userRepository
-            .findOneOrFail({
-                where: { id },
-                relations: { userAvatar: true }
-            })
-            .catch(() => {
+        return this.userRepository.findOneByOrFail({ id }).catch(error => {
+            if (error instanceof EntityNotFoundError)
```

Comment:
I'm not sure about this. It's technically okay but context of calling fuction is wrong. Because it's only used in `validate` function. So in this case `EntityNotFoundError` could equals `Unauthorized`. Consider renaming function or other resolution

Reply:
Ye, but that's the same case as in username, we talking about that, i leave here notFound, because we use username as nick, i can here return notFound because the context is for username, on password i return unauthorize.

In real product companies, didn't return something like not found, because this help with security increases.

I can change it for Unautorize, because this is in ```validate``` only.

Reply:
I change Exception and name, let's see in new commit

Reply:
done